src/classes/frSetupController.cls

```diff
@@ -157,20 +268,51 @@ public with sharing class frSetupController {
 		List<frMapping__c> upsertList = new List<frMapping__c>();
 		List<frMapping__c> allMappings = new List<frMapping__c>(donationMappings);
 		allMappings.addAll(donorMappings);
+		allMappings.addAll(accMappings);
+		allMappings.addAll(giftTransactionMappings);
+		allMappings.addAll(giftCommitmentMappings);
 		for(frMapping__c mapping : allMappings) {
+
 			if(String.isNotBlank(mapping.sf_Name__c)) {
-                mapping.Name = (mapping.Type__c == DONATION_TYPE ? 'o.' : 'c.') + mapping.sf_Name__c;
+                //mapping.Name = (mapping.Type__c == DONATION_TYPE ? 'o.' : 'c.') + mapping.sf_Name__c;
+				if(mapping.Type__c == DONATION_TYPE) mapping.Name = 'o.' + mapping.sf_Name__c;
+				if(mapping.Type__c == DONOR_TYPE) mapping.Name = 'c.' + mapping.sf_Name__c;
+				if(mapping.Type__c == PERSON_ACCOUNT_TYPE) mapping.Name = 'a.' + mapping.sf_Name__c;
+				if(mapping.Type__c == GIFT_TRANSACTION_TYPE) mapping.Name = 't.' + mapping.sf_Name__c;
+				if(mapping.Type__c == GIFT_COMMITMENT_TYPE) mapping.Name = 's.' + mapping.sf_Name__c;
 				if(mapping.Name.length() > customSettingNameLength) {
 					mapping.Name = mapping.Name.subString(0, customSettingNameLength);
 				}
 				upsertList.add(mapping);
 			}
 		}
-
-		Schema.DescribeSObjectResult describe = frMapping__c.sObjectType.getDescribe();
-		if(upsertList.size() > 0 && describe.isCreateable() && describe.isUpdateable()) {
-			upsert upsertList;
-			addConfirm('Mappings saved successfully');
+		Boolean found = false;
+		for(frMapping__c mapping : accMappings) {
+			if (mapping.sf_Name__c == 'fr_ID__c') {
+				found = true;
+				break;
+			}
+		}
+		for(frMapping__c mapping : giftTransactionMappings) {
+			if (mapping.sf_Name__c == 'fr_ID__c') {
+				found = true;
+				break;
+			}
+		}
+		for(frMapping__c mapping : giftCommitmentMappings) {
+			if (mapping.sf_Name__c == 'fr_ID__c') {
+				found = true;
+				break;
+			}
+		}
+		if(found || isNonProfitOrg != true) {
+			Schema.DescribeSObjectResult describe = frMapping__c.sObjectType.getDescribe();
+			if(upsertList.size() > 0 && describe.isCreateable() && describe.isUpdateable()) {
+				upsert upsertList;
+				addConfirm('Mappings saved successfully');
+			}
+		} else {
+			addError('fr_ID__c Field must be mapped and present on all NPC Objects.');
```

Comment:
I think we don't want to enforce that users map the Funraise ID field themselves, as since we know it is required we should handle mapping ourselves in the logic that saves the records, similar to how it works today for Opportunites & Contacts

Can we replace this with validation that the field exists (since, as we previously discussed we won't be able to create the custom fields on the NPC objects, we'll need a way to ensure the custom fields we need are there)

Reply:
I want to follow up on this one as well, let me know any thoughts!

Reply:
We could do this but im worried since we cant create the fields with the package that without a way in setup to warn/notify the user that the fields dont exist it could create confusion when the package isnt working as expected. 

I could replace it with validation that the field exists and have a warning banner on the setup page if they dont, would this be better?

Reply:
I'm fine with validation that the field exists, but once it exists we should not be forcing the user to map to values it